client.go

```diff
@@ -40,34 +40,39 @@ type Client interface {
 // ResourceClient allows access to resource-specific clients
 type ResourceClient interface {
 	// Organization gets the OrganizationClient for the specific top-level organization, or user account.
-	// ErrNotTopLevelOrganization will be returned if the organization is not top-level
+	//
+	// ErrNotTopLevelOrganization will be returned at usage time if the organization is not top-level.
 	Organization(o OrganizationRef) OrganizationClient
 
-	// Organizations returns the OrganizationsClient handling sets of organizations
+	// Organizations returns the OrganizationsClient handling sets of organizations.
 	Organizations() OrganizationsClient
 
-	// Repository gets the RepositoryClient for the specified RepositoryRef
+	// Repository gets the RepositoryClient for the specified RepositoryRef.
 	Repository(r RepositoryRef) RepositoryClient
 
-	// Repositories returns the RepositoriesClient handling sets of organizations
+	// Repositories returns the RepositoriesClient handling sets of organizations.
 	Repositories() RepositoriesClient
 }
 
-// OrganizationsClient operates on organizations the user has access to
+// OrganizationsClient operates on organizations the user has access to.
 type OrganizationsClient interface {
-	// Get a specific organization the user has access to
-	// This might also refer to a sub-organization
-	// ErrNotFound is returned if the resource does not exist
+	// Get a specific organization the user has access to.
+	// This might also refer to a sub-organization.
+	//
+	// ErrNotFound is returned if the resource does not exist.
 	Get(ctx context.Context, o OrganizationRef) (*Organization, error)
 
-	// List all top-level organizations the specific user has access to
-	// List should return all available organizations, using multiple paginated requests if needed
+	// List all top-level organizations the specific user has access to.
+	//
+	// List should return all available organizations, using multiple paginated requests if needed.
 	List(ctx context.Context) ([]Organization, error)
 
-	// Children returns the immediate child-organizations for the specific IdentityRef o.
-	// The IdentityRef may point to any sub-organization that exists
-	// This is not supported in GitHub
-	// Children should return all available organizations, using multiple paginated requests if needed
+	// Children returns the immediate child-organizations for the specific OrganizationRef o.
+	// The OrganizationRef may point to any sub-organization that exists.
+	//
+	// This is not supported in GitHub.
```

Comment:
This is a provider-specific note, do we want to have them here?

Reply:
IDK to be honest, but I think it's okay if they are few, and notable, like in this case.
Might revisit this later if they grow too many or complex.